My suspicions have been confirmed locally (finally!):

```
[INFO]  passing in data object to featureEvent!
[WARN]  TypeConverter: jsObjectToJavaKrollDict converting object...
[WARN]  TypeConverter: isNumber == true!
[WARN]  TypeConverter: isString == true!
[ERROR] JNIUtil: jsObjectToJavaKrollDict key: java.lang.Integer
[WARN]  W/System.err: java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.String
[WARN]  W/System.err: 	at org.appcelerator.titanium.util.TiConvert.toJSON(TiConvert.java:723)
[WARN]  W/System.err: 	at ti.modules.titanium.analytics.AnalyticsModule.featureEvent(AnalyticsModule.java:98)
[WARN]  W/System.err: 	at org.appcelerator.kroll.runtime.v8.V8Object.nativeFireEvent(Native Method)
[WARN]  W/System.err: 	at org.appcelerator.kroll.runtime.v8.V8Object.fireEvent(V8Object.java:62)
[WARN]  W/System.err: 	at org.appcelerator.kroll.KrollProxy.doFireEvent(KrollProxy.java:872)
[WARN]  W/System.err: 	at org.appcelerator.kroll.KrollProxy.handleMessage(KrollProxy.java:1095)
[WARN]  W/System.err: 	at org.appcelerator.titanium.proxy.TiViewProxy.handleMessage(TiViewProxy.java:354)
[WARN]  W/System.err: 	at android.os.Handler.dispatchMessage(Handler.java:98)
[WARN]  W/System.err: 	at android.os.Looper.loop(Looper.java:148)
[WARN]  W/System.err: 	at android.app.ActivityThread.main(ActivityThread.java:5417)
[WARN]  W/System.err: 	at java.lang.reflect.Method.invoke(Native Method)
[WARN]  W/System.err: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
[WARN]  W/System.err: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
[ERROR] TiExceptionHandler: (main) [43784,43784] ----- Titanium Javascript Runtime Error -----
[ERROR] TiExceptionHandler: (main) [0,43784] - In /alloy/controllers/index.js:16,35
[ERROR] TiExceptionHandler: (main) [0,43784] - Message: Uncaught java.lang.Integer cannot be cast to java.lang.String
[ERROR] TiExceptionHandler: (main) [0,43784] - Source:         var result = Ti.Analytics.featureEvent("levels", data);
[ERROR] V8Exception: Exception occurred at /alloy/controllers/index.js:16: Uncaught java.lang.Integer cannot be cast to java.lang.String
[ERROR] V8Exception: java.lang.Integer cannot be cast to java.lang.String
```

--- a/android/runtime/v8/src/native/TypeConverter.cpp
+++ b/android/runtime/v8/src/native/TypeConverter.cpp
@@ -663,20 +663,18 @@ jobject TypeConverter::jsValueToJavaObject(v8::Isolate* isolate, JNIEnv *env, v8
 
 			for (int i = 0; i < numKeys; i++) {
 				v8::Local<v8::Value> jsObjectPropertyKey = objectKeys->Get((uint32_t) i);
-				bool keyIsNew, valueIsNew;
-				jobject javaObjectPropertyKey = TypeConverter::jsValueToJavaObject(isolate, env, jsObjectPropertyKey, &keyIsNew);
+				bool valueIsNew;
+				jstring javaStringPropertyKey = TypeConverter::jsValueToJavaString(isolate, env, jsObjectPropertyKey);
 				v8::Local<v8::Value> jsObjectPropertyValue = jsObject->Get(jsObjectPropertyKey);
 				jobject javaObjectPropertyValue = TypeConverter::jsValueToJavaObject(isolate, env, jsObjectPropertyValue, &valueIsNew);
 
 				jobject result = env->CallObjectMethod(javaHashMap,
 				                                       JNIUtil::hashMapPutMethod,
-				                                       javaObjectPropertyKey,
+				                                       javaStringPropertyKey,
 				                                       javaObjectPropertyValue);
 				env->DeleteLocalRef(result);
+				env->DeleteLocalRef(javaStringPropertyKey);
 
-				if (keyIsNew) {
-					env->DeleteLocalRef(javaObjectPropertyKey);
-				}
 				if (valueIsNew) {
 					env->DeleteLocalRef(javaObjectPropertyValue);
 				}
@@ -715,20 +713,18 @@ jobject TypeConverter::jsObjectToJavaKrollDict(v8::Isolate* isolate, JNIEnv *env
 
 		for (int i = 0; i < numKeys; i++) {
 			v8::Local<v8::Value> jsObjectPropertyKey = objectKeys->Get((uint32_t) i);
-			bool keyIsNew, valueIsNew;
-			jobject javaObjectPropertyKey = TypeConverter::jsValueToJavaObject(isolate, env, jsObjectPropertyKey, &keyIsNew);
+			bool valueIsNew;
+			jstring javaStringPropertyKey = TypeConverter::jsValueToJavaString(isolate, env, jsObjectPropertyKey);
 			v8::Local<v8::Value> jsObjectPropertyValue = jsObject->Get(jsObjectPropertyKey);
 			jobject javaObjectPropertyValue = TypeConverter::jsValueToJavaObject(isolate, env, jsObjectPropertyValue, &valueIsNew);
 
 			jobject result = env->CallObjectMethod(javaKrollDict,
 				                                   JNIUtil::krollDictPutMethod,
-				                                   javaObjectPropertyKey,
+				                                   javaStringPropertyKey,
 				                                   javaObjectPropertyValue);
 			env->DeleteLocalRef(result);
+			env->DeleteLocalRef(javaStringPropertyKey);
 
-			if (keyIsNew) {
-				env->DeleteLocalRef(javaObjectPropertyKey);
-			}
 			if (valueIsNew) {
 				env->DeleteLocalRef(javaObjectPropertyValue);
 			}

